### DISCUSSION TITLE
About "-clip-path" option for convert command

### DISCUSSION THREAD

**Discussion by coffeesmell** [ImageMagick/ImageMagick#3943] [Help]
Is there a document about how to use "-clip-path" option during the convert command?

I saw some sample commands with this option supplying only number for the {id}

Example:.
`-alpha transparent -clip-path #1 -alpha opaque +clip-path #1`

could a clipping path name be used? if so, does image magick supports japanese clipping path names for this option?

Example:.
`-alpha transparent -clip-path "PathX" -alpha opaque +clip-path "PathX"`

`-alpha transparent -clip-path "例えば" -alpha opaque +clip-path "例えば"`

**Comment by snibgo** [comment]
See https://imagemagick.org/script/command-line-options.php#clip-path

In the example you show, "#1" is the name of the clip-path (also known as the "id" of the clip-path). "magick identify -verbose yourfile.ext" will show you clip-paths with their names.

I suppose IM accepts UTF8 characters for the clip-path name.

**Reply by coffeesmell** [reply]
hi snibgo,

Thank you for the clarification.

In case where there are more than 1 clipping paths, can Imagemagick identify which is the main clipping path? 

Usually this is the case for photoshop images EPS and PSD, where there are multiple clipping paths but the main clipping path can be set.

**Comment by snibgo** [comment]
> In case where there are more than 1 clipping paths, can Imagemagick identify which is the main clipping path?

I don't think IM knows about a "main" clipping path. "-clipping-path id" needs a name or number eg "#1". Or you can use "-clip" with no ID, which seems to be the same as "-clipping-path #1", which takes the first clipping path.
